lib/SILOptimizer/Analysis/MemoryBehavior.cpp

```diff
@@ -56,6 +56,8 @@ class MemoryBehaviorVisitor
   /// The value we are attempting to discover memory behavior relative to.
   SILValue V;
 
+  Optional<SILValue> cachedValueAddress;
```

Comment:
You don't need to use Optional for SILValue. SILValue has an "invalid" state.

Reply:
oops, not thinking